cli/src/pcluster/utils.py

```diff
@@ -926,6 +930,20 @@ def get_info_for_amis(ami_ids):
         error(e.response.get("Error").get("Message"))
 
 
+def get_ami_id(parent_image):
+    """Get ami id from parent image."""
+    if parent_image.startswith("arn"):
+        ami_id = get_info_for_ami_from_arn(parent_image).get("image").get("outputResources").get("amis")[0].get("image")
```

Comment:
I think we should hide this series of getters in the ImageBuilderClient() by creating a method like:
`ImageBuilderClient().get_image_id(image_arn)`

because with the current code the caller (get_ami_id function) need to know all the structure of the boto3 answer, instead this information should be encapsulated in the ImageBuilderClient.

If needed in future we can have other methods exposed by the Client to have other information.

The other reason is that you should check for the fields in the response.
by doing `.get("image").get("outputResources").get("amis")[0].get("image")` you can get some KeyError if the ami is not found or the output format is not the expected one.

In the ImageBuilderClient().get_image_id you can do something like:
`.get("image", []).get("outputResources", []).get("amis", [])` by managing and checking all the errors and then return an empty string if the ami is not found.

Reply:
Done